tests/output/webpage/test_utils.js

```diff
@@ -116,23 +116,36 @@ var warningTest = function(title, code, warnings) {
 
 var assertTest = function(options) {
     options.test = function(output, errors, testResults, callback) {
-        if (!options.reason) {
-            expect(errors.length).to.be.equal(0);
-        } else {
-            if (options.fromTests) {
+        // If the test is checking the results from validation tests:
+        if (options.fromTests) {
+            if (!options.reason) {
+                expect(errors.length).to.be.equal(0);
+                expect(testResults).to.not.equal([]);
+                expect(testResults[0].state).to.not.equal("fail");
+            } else {
                 expect(testResults).to.not.equal([]);
                 expect(testResults[0].state).to.be.equal("fail");
-                expect(testResults[0].results[0].meta.alsoMessage)
-                    .to.be.equal(options.reason);
+                // If specific reason given, verify it matches:
+                if (options.reason !== "fail") {
+                    expect(testResults[0].results[0].meta.alsoMessage)
+                        .to.be.equal(options.reason);
```

Comment:
I guess if we get an error here the `expect` will still fail since we won't get the right `testResults`.